agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -71,8 +73,12 @@ def weave_attach(container)
 
     # @param [String] container_id
     # @param [String] overlay_cidr
-    def attach_overlay(container_id, overlay_cidr)
-      network_adapter.exec(['--local', 'attach', overlay_cidr, '--rewrite-hosts', container_id])
+    def attach_overlay(container)
+      if container.overlay_suffix != OVERLAY_SUFFIX
```

Comment:
Yes, this should ideally check that the `container.overlay_addr.network` from the container label overlaps with the expected grid subnet, such that the existing `container.overlay_addr` is within the new grid subnet.

For 0.16 -> 0.17 it can be implicitly assumed that both of those are within the `10.81.0.0/#{OVERLAY_SUFFIX}`. With #1323 the `WeaveWorker` will need access to the `node_info['grid']['subnet']`, and better `IPAddr` logic for this comparison.